contracts/ERC20MultiDelegate.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Votes.sol";
+
+/**
+ * @dev A child contract which will be deployed by the ERC20MultiDelegate utility contract
+ * This is a proxy delegator contract to vote given delegatee on behalf of original delegator
+ */
+contract ERC20ProxyDelegator {
+    constructor(ERC20Votes _token, address _delegatee) {
+        _token.approve(msg.sender, type(uint256).max);
+        _token.delegate(_delegatee);
+    }
+}
+
+/**
+ * @dev A utility contract to let delegators to pick multiple delegatee
+ */
+contract ERC20MultiDelegate is ERC1155, Ownable {
+    using Address for address;
+
+    ERC20Votes token;
+
+    /**
+     * @dev Constructor.
+     * @param _token The ERC20 token address
+     */
+    constructor(ERC20Votes _token) ERC1155("http://some.metadata.url/{id}") {
+        token = _token;
+    }
+
+    /**
+     * @dev Public method for the delegation of multiple delegatees.
+     * @param delegatees List of delegatee addresses
+     * @param amounts ERC20 voting power amount to be distributed among delegatees
+     */
+    function delegateMulti(
```

Comment:
Either this should be called `deposit` or `withdraw` should be called `undelegateMulti`. Personally I think `depositMulti/withdrawMulti` makes sense, with corresponding convenience functions that only deposit and assign to a single address.